### ISSUE TITLE
ERROR getting assigned IP of eth0 using wireguard mullvad

### ISSUE THREAD

**Issue by Fastjur** [qdm12/gluetun#1591]
(Sorry, it seems that there is a template for bug reports, but that did not show when converting a discussion to an issue obviously, let me know if you want me to resubmit properly).

It seems that mullvad may have had some changes that have broken many clients.
I am unsure if the discussion https://github.com/qdm12/gluetun/discussions/1568 for OpenVPN relates to the setup using wireguard, but I nothing has changed in my setup and gluetun has suddenly stopped working overnight when using Mullvad with a wireguard setup.

### Discussed in https://github.com/qdm12/gluetun/discussions/1590

<div type='discussions-op-text'>

<sup>Originally posted by **Fastjur** May 22, 2023</sup>
Hi all,

A couple of days ago I have set up gluetun in a docker container to provide a VPN to some of my other docker containers.
I worked as expected for a couple of days, but suddenly has stopped working.
The logs from gluetun are not really helpful, and I could not find anything about this:

```
========================================
========================================
=============== gluetun ================
========================================
=========== Made with ❤️ by ============
======= https://github.com/qdm12 =======
========================================
========================================

Running version latest built on 2023-05-22T06:07:02.257Z (commit eed62fd)

🔧 Need help? https://github.com/qdm12/gluetun/discussions/new
🐛 Bug? https://github.com/qdm12/gluetun/issues/new
✨ New feature? https://github.com/qdm12/gluetun/issues/new
☕ Discussion? https://github.com/qdm12/gluetun/discussions/new
💻 Email? quentin.mcgaw@gmail.com
💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
2023-05-22T10:21:13+02:00 ERROR getting assigned IP of eth0: IP address not found for interface: interface eth0 in 1 addresses
2023-05-22T10:21:13+02:00 INFO Shutdown successful
```

The VPN provider is mullvad, and I have double checked that I have enough time left at mullvad (it up to half june for now).
I have also recreated the mullvad private key and configuration (I downloaded a new config zip).
I did notice that the IP address changed, which is odd to me, but the new IP still does not work.

Here is my docker compose for gluetun (don't mind the line numbers).
I have removed the wireguard address, not sure of that is needed, but it's just the ipv4 address in cidr notation from the mullvad config file.

```yaml
 63   gluetun:
 64     image: ghcr.io/qdm12/gluetun
 65     container_name: gluetun
 66     # line above must be uncommented to allow external containers to connect. See https://github.com/qdm12/gluetun/wiki/Connect-a-container-to-gluetun#external-container-to-gluetun
 67     cap_add:
 68       - NET_ADMIN
 69     devices:
 70       - /dev/net/tun:/dev/net/tun
 71     networks:
 72       - traefik-net
 73     ports:
 74       - 8888:8888/tcp # HTTP proxy
 75       - 8388:8388/tcp # Shadowsocks
 76       - 8388:8388/udp # Shadowsocks
 77       - 8080:8080 # qbt webui
 78       - 8999:8999 # qbt
 79       - 8999:8999/udp # qbt
 80       - 3000:3000 # firefox container http
 81 #      - 3001:3001 # firefox container https
 82     volumes:
 83       - ./gluetun:/gluetun
 84     environment:
 85       # See https://github.com/qdm12/gluetun/wiki
 86       - VPN_SERVICE_PROVIDER=mullvad
 87       - VPN_TYPE=wireguard
 88       - SHADOWSOCKS_PASSWORD=${SHADOWSOCKS_PASSWORD}
 89       # OpenVPN:
 90       #      - OPENVPN_USER=
 91       #      - OPENVPN_PASSWORD=
 92       # Wireguard:
 93       - WIREGUARD_PRIVATE_KEY=${MULLVAD_VPN_WIREGUARD_PRIVATEKEY}
 94       - WIREGUARD_ADDRESSES=<ipv4/cidr notation from the wireguard config file>
 95       - SERVER_CITIES=Amsterdam
 96       # Timezone for accurate log times
 97       - TZ=Europe/Amsterdam
 98       # Server list updater. See https://github.com/qdm12/gluetun/wiki/Updating-Servers#periodic-update
 99       - UPDATER_PERIOD=24h
100   #      - UPDATER_VPN_SERVICE_PROVIDERS=
101     labels:
102       - "traefik.enable=true"
103       - "traefik.docker.network=traefik-net"
104
105       - "traefik.http.services.qbittorrentvpn.loadbalancer.server.port=8080"
106       - "traefik.http.routers.qbittorrentvpn.rule=Host(`qbt.${DOMAIN}`)"
107       - "traefik.http.routers.qbittorrentvpn.entrypoints=websecure"
108       - "traefik.http.routers.qbittorrentvpn.service=qbittorrentvpn"
109       - "traefik.http.routers.qbittorrentvpn.tls.certresolver=letsencryptresolver"
110
111       - "traefik.http.services.firefoxvpn.loadbalancer.server.port=3000"
112       - "traefik.http.routers.firefoxvpn.rule=Host(`firefoxvpn.${DOMAIN}`)"
113       - "traefik.http.routers.firefoxvpn.entrypoints=websecure"
114       - "traefik.http.routers.firefoxvpn.service=firefoxvpn"
115       - "traefik.http.routers.firefoxvpn.tls.certresolver=letsencryptresolver"
```

**Comment by Fastjur** [comment]
I think there has already been a related discussion regarding Mullvad and OpenVPN: https://github.com/qdm12/gluetun/discussions/1568

**Comment by WinstonGFX** [comment]
Same issue here

**Comment by Fastjur** [comment]
Hey @WinstonGFX, I just pulled the latest image and it seems to be fixed for me.
If you're using docker compose run:

```bash
docker compose pull
docker compose up -d
```

I'm closing the issue for now, it seems there has been a fix.

**Comment by qdm12** [comment]
Yes this was fixed in #1588 sorry for the disturbance 😉

**Comment by WinstonGFX** [comment]
Thank you both - just pulled the latest image and things appear to be back up and running

**Comment by Fastjur** [comment]
> Yes this was fixed in #1588 sorry for the disturbance 😉

No worries! You fixed it extremely quick haha, thanks :)
